pkg/proxy/healthcheck/service_health.go

```diff
@@ -163,15 +162,23 @@ func (h hcHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	} else {
 		resp.WriteHeader(http.StatusOK)
 	}
-	fmt.Fprintf(resp, strings.Trim(dedent.Dedent(fmt.Sprintf(`
-		{
-			"service": {
-				"namespace": %q,
-				"name": %q
-			},
-			"localEndpoints": %d
-		}
-		`, h.name.Namespace, h.name.Name, count)), "\n"))
+	info := serviceInfo{
```

Comment:
I'm curious if there could be an advantage to using [anonymous structs](https://talks.golang.org/2012/10things.slide#2) instead of introducing custom struct types (like `serviceInfo`) which are only used in one location.

For example, this code could use anonymous structs:
```go
info := struct{
    Service map[string]string `json:"service"`
    ...
}{
    Service: map[string]string{
        ...
    },
    LocalEndpoints: count,
}
patchBytes, err := json.Marshal(&info)
```

I'm not entirely sure replacing named struct types with anonymous structs is the best choice throughout this PR, but I do think it could be useful in multiple places.

What do you think?